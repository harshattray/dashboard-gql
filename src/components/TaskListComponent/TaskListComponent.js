/**
 * @Author: harsha
 * @Date:   2019-05-20T10:59:37+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-05-20T23:32:01+05:30
 */

import React, { Component, Fragment } from "react";
import {
  Form,
  Card,
  Button,
  Segment,
  Dimmer,
  Loader,
  Feed,
  Icon
} from "semantic-ui-react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { reduxForm, Field, formValueSelector, reset } from "redux-form";
import { RenderCheckBox } from "../RenderCheckBoxComponent/RenderCheckBoxComponent";
import { updateTask } from "../../actions/TaskActions";

export class TaskListComponent extends Component {
  render() {
    const { tasksInfo, updateTask, handleSubmit } = this.props;
    return (
      <Fragment>
        <RenderTaskList
          tasksInfo={tasksInfo}
          updateTask={updateTask}
          handleSubmit={handleSubmit}
        />
      </Fragment>
    );
  }
}

export const RenderTaskList = ({ tasksInfo, updateTask, handleSubmit }) => {
  const listRenderer = tasksInfo.map(data => {
    return (
      <div className="list-view" key={data.id}>
        <Feed>
          <Feed.Event>
            <Feed.Content>
              Status:
              {data.completed ? (
                <Icon name="thumbs up" color="green" />
              ) : (
                <Icon name="thumbs down" color="red" />
              )}
              <Feed.Summary>{data.comment}</Feed.Summary>
              <Form name="taskListForm" noValidate>
                <div className="row">
                  <Field
                    name="taskCheckBox"
                    component={RenderCheckBox}
                    label="Change Status"
                    updateTask={updateTask}
                    handleSubmit={handleSubmit}
                    taskID={data.id}
                  />
                </div>
              </Form>
            </Feed.Content>
          </Feed.Event>
        </Feed>
      </div>
    );
  });
  return <Fragment>{listRenderer}</Fragment>;
};

function mapStateToProps({ carStack, form, tasksStack }) {
  return {
    tasksInfo: tasksStack.tasksList.tasks,
    isFetchingTasks: tasksStack.isFetchingTasks
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ updateTask }, dispatch);
}
const afterSubmitdata = (result, dispatch) => dispatch(reset("taskListForm"));

TaskListComponent = reduxForm({
  form: "taskListForm",
  destroyOnUnmount: false,
  onSubmitSuccess: afterSubmitdata
})(TaskListComponent);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TaskListComponent);
