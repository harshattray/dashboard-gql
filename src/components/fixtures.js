/**
 * @Author: harsha
 * @Date:   2019-05-20T23:28:57+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-05-21T00:02:14+05:30
 */
export const listProps = {
  fetchTasks: jest.fn(),
  isFetchingTasks: false,
  openModal: jest.fn(),
  tasksInfo: {
    tasks: [
      {
        id: "710ef490-7b26-11e9-9bc3-29da9ae12ff0",
        taskType: "ADD_DOCUMENT",
        comment: "Passport",
        completed: false
      }
    ]
  }
};

export const tasksInfo = {
  handleSubmit: jest.fn(),
  updateTask: jest.fn(),
  tasksInfo: {
    tasks: [
      {
        id: "710ef490-7b26-11e9-9bc3-29da9ae12ff0",
        taskType: "ADD_DOCUMENT",
        comment: "Passport",
        completed: false
      }
    ]
  }
};
