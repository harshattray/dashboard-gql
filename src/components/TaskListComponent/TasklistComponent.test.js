/**
 * @Author: harsha
 * @Date:   2019-05-20T23:32:45+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-05-20T23:46:16+05:30
 */
import React from "react";
import { shallow } from "enzyme";
import { TaskListComponent, RenderTaskList } from "./TaskListComponent";
import { tasksInfo } from "../fixtures";

const props = { tasksInfo };

describe("TaskListComponent", () => {
  let wrapper = shallow(<TaskListComponent {...props} />);

  it("contains RenderTaskList", () => {
    expect(wrapper.find(RenderTaskList)).toMatchSnapshot();
  });

  it("expect to render TasksList", () => {
    console.log(wrapper.debug());
    expect(wrapper.find("tasksInfo").length).toBeGreaterThanOrEqual(0);
  });
});
