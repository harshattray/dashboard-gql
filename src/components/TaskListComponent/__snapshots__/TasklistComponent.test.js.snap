// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListComponent contains RenderTaskList 1`] = `ShallowWrapper {}`;
